### ISSUE TITLE
Question: Can you use go-dnsmasq as a sidecar in a pod in Kubernetes?

### ISSUE THREAD

**Issue by johanhaleby** [janeczku/go-dnsmasq#11]
I'd like to configure an nginx resolver to point to go-dnsmasq, my nginx config looks like this:

```
location ~ ^/(.+)$ {
        resolver 127.0.0.1:53;
        set $backend "http://$1:80";
        proxy_pass $backend;
}
```

and my Kubernetes pod specification looks like this:

``` yaml
spec:
  containers:
    - name: nginx
      image: "nginx:1.10.0"
      ports:
        - containerPort: 8080
          name: "external"
          protocol: "TCP"
    - name: dnsmasq
      image: "janeczku/go-dnsmasq:release-1.0.5"
      args:
        - --listen
        - "0.0.0.0:53"
```

But I run into this error:

```
[error] 9#9: *734 myservice could not be resolved (2: Server failure), client: 127.0.0.1, server: nginx-proxy, request: "GET /myservice HTTP/1.1", host: "localhost:8080"
127.0.0.1 - xxx [30/May/2016:10:34:23 +0000] "GET /myservice HTTP/1.1" 502 173 "-" "curl/7.38.0" "-"
```

But I can see that go-dnsmasq is up and running:

``` bash
PORT     STATE SERVICE
53/tcp   open  domain
8080/tcp open  http-proxy
53/udp   open  domain
```

Is this suppose to work or do I have to manually derive an nginx image based on for example [alpine-kubernetes](https://hub.docker.com/r/janeczku/alpine-kubernetes/)? I think "my" approach would be more [simple](https://www.infoq.com/presentations/Simple-Made-Easy) and it would allow me to use the vanilla nginx image. Do you know if this should work or what I could be doing wrong (if you think it's correctly configured then it might be an nginx issue and I'm of course not implying that you should help me out with that)?

I've also posted this question on [stackoverflow](http://stackoverflow.com/questions/37523615/dynamic-proxy-pass-in-nginx-to-another-pod-in-kubernetes).

**Comment by janeczku** [comment]
Hi @johanhaleby,

Your pod spec should work the way you intended. Go-dnsmasq seems to return a `SERVFAIL` response. Can you add the `--verbose` flag to go-dnsmasq args in your spec and post the log?

**Comment by johanhaleby** [comment]
@janeczku Thanks for your reply. I've now change the pod spec to:

``` yaml
- name: dnsmasq
  image: "janeczku/go-dnsmasq:release-1.0.5"
  args:
    - --listen
    - "127.0.0.1:53"
    - --default-resolver
    - --append-search-domains
    - --hostsfile=/etc/hosts
    - --verbose
```

since I tracked this down in a [run script](https://github.com/janeczku/docker-alpine-kubernetes/blob/master/rootfs/etc/services.d/resolver/run) in alpine-kubernetes but it still won't work. Here's the log:

```
level=debug msg="Received DNS query for \"my-service.\" from \"127.0.0.1:45314\" with type 1"
level=debug msg="Querying nameserver 10.247.240.10:53 question my-service.default.svc.cluster.local."
level=debug msg="Querying nameserver 10.247.240.10:53 question my-service.svc.cluster.local."
level=debug msg="Querying nameserver 10.247.240.10:53 question my-service.cluster.local."
level=debug msg="Querying nameserver 10.247.240.10:53 question my-service.c.my-cluster.internal."
level=debug msg="Querying nameserver 10.247.240.10:53 question my-service.google.internal."
```

Does this look ok?

Nginx now gives me another error:

```
2016/05/31 11:50:13 [error] 6#6: *37 my-service could not be resolved (3: Host not found), client: 127.0.0.1, server: nginx, request: "GET /my-service HTTP/1.1", host: "localhost:8080"
127.0.0.1 - johanhaleby [31/May/2016:11:50:13 +0000] "GET /my-service HTTP/1.1" 502 173 "-" "curl/7.38.0" "-"
```

I.e. I seem to get a "Host not found error". Fwiw the `/etc/resolv.conf` file in the `go-dnsmasq` container looks like this:

```
nameserver 127.0.0.1 # added by go-dnsmasq
search default.svc.cluster.local svc.cluster.local cluster.local c.my-cluster.internal. google.internal.
# nameserver 10.247.240.10
options ndots:5
```

and `/etc/hosts` looks like this:

```
# Kubernetes-managed hosts file.
127.0.0.1   localhost
::1 localhost ip6-localhost ip6-loopback
fe00::0 ip6-localnet
fe00::0 ip6-mcastprefix
fe00::1 ip6-allnodes
fe00::2 ip6-allrouters
10.244.0.15 nginx-2470195802-5hpm1
```

**Comment by johanhaleby** [comment]
I managed to solve this (with the new kubernetes pod specification) and when I disabled the ipv6 as resolver: `resolver 127.0.0.1:53 ipv6=off;`

Thanks for your help!
